native/src/snappyclient/cpp/types/Decimal.cpp

```diff
@@ -360,15 +360,16 @@ size_t Decimal::toString(std::string& str) const {
   char buf[thrift::snappydataConstants::DECIMAL_MAX_PRECISION + 4];
   char* bufp = buf;
   io::snappydata::client::impl::FreePointer freep(0);
-  const size_t ndigits = mpz_sizeinbase(m_bigInt, 10);
+  size_t ndigits = mpz_sizeinbase(m_bigInt, 10);
   if (ndigits > 128) {
     bufp = new char[ndigits + 2];
     freep.reset(bufp);
   }
   mpz_get_str(bufp, 10, m_bigInt);
-
+  ndigits = strlen(bufp);// to get exact size
   // now the three cases of '.' inside, before and not at all
   const bool neg = (*bufp == '-');
+  if (neg) ndigits--;
```

Comment:
Alternative would be to not add "neg" while returning the size?

Reply:
Guess its specified to add one space / character in case its a negative number.

Reply:
mpz_get_str (line: 368) fill "bufp" with a negative sign only if the value is negative. So need to remove length by 1

Reply:
> Alternative would be to not add "neg" while returning the size?
Yes, it should not be added while returning the size, but for this  mpz_sizeinbase should return the correct values all the time. I guess at the time of development, neg has been added with the expectation that  mpz_sizeinbase returns the correct size all the time. Now, after current changes, neg should be added, because we are doing (ndigit --) reducing size by 1 for neg. It wont reduce size then it returns incorrect "wholeDigits", which was causing the issue.